src/createSigningPayload.ts

```diff
@@ -4,7 +4,40 @@ import { Options, UnsignedTransaction } from './util';
  * Construct the signing payload from an unsigned transaction and export it to
  * a remote signer (this is often called "detached signing").
  *
+ * **Important!** The registry needs to be passed into the `options` argument.
+ * This registry needs to be updated with latest metadata, so before calling
+ * this function, make sure to run `registry.setMetadata(metadata)` first.
+ *
+ * **Important!** The return value of this function is **NOT** the actual
+ * payload to sign: the actual payload to sign includes `method` which should
+ * not be length-prefixed. To construct the actual payload to sign, see the
+ * example.
+ *
  * @param unsigned - The JSON representing the unsigned transaction.
+ * @param options - Registry and metadata used for constructing the method.
+ *
+ * @example
+ * ```ts
+ * // Serialized signing payload.
+ * const signingPayload = createSigningPayload(unsigned, {
+ *   metadataRpc,
+ *   registry
+ * });
+ *
+ * // Construct an `ExtrinsicPayload` class.
+ * const extrinsicPayload = registry
+ *   .createType('ExtrinsicPayload', unsigned, {
+ *     version: unsigned.version,
```

Comment:
There's been a lot of confusion about `EXTRINSIC_VERSION` vs. `transactionVersion`. Maybe another note to clarify what is being used here?